### PR TITLE
Add support for a new node type: FormatterFilter

### PR DESCRIPTION
**Pull request by jimlambrt**
add support for FormatterFilter which will format an event and then allow you to filter events based on the formatter's struct vs the event struct

### CODE REVIEW

File: formatter_filters/cloudevents/formatter_filter.go [hunk 1]
```diff
@@ -106,11 +110,13 @@ func (f *Formatter) validate() error {
 }
 
 // Process formats the Event as a cloudevent and stores that formatted data in
-// Event.Formatted with a key of "cloudevents-json" (cloudevents.FormatJSON)
-func (f *Formatter) Process(ctx context.Context, e *eventlogger.Event) (*eventlogger.Event, error) {
-	const op = "cloudevents.(Formatter).Process"
+// Event.Formatted with a key of "cloudevents-json" (cloudevents.FormatJSON).
```
Comment (s-christoff) [inline formatter_filters/cloudevents/formatter_filter.go:113]: Dropped this
```
// Process formats the Event as a cloudevent and stores that formatted data in
```

File: filters/encrypt/errors.go [hunk 2]
```diff
@@ -4,7 +4,5 @@ import (
 	"errors"
 )
 
-var (
-	// ErrInvalidParameter defines a value for invalid parameter errors
-	ErrInvalidParameter = errors.New("invalid parameter")
-)
+// ErrInvalidParameter defines a value for invalid parameter errors
```
Comment (s-christoff) [inline filters/encrypt/errors.go:7]: What is the benefit or downside of changing this?

**Review by s-christoff — APPROVED** [review]
One comment about a comment that was left out, otherwise looks great!

File: formatter_filters/cloudevents/formatter_filter.go (hunk 1 above)
Comment (jimlambrt) [reply formatter_filters/cloudevents/formatter_filter.go:113]: I think, it's in the line above

**Pull request merged**